### PR TITLE
Add ability to create run from YAML [monaco]

### PR DESCRIPTION
**Pull request by marniks7**
# Changes

Add ability to create PipelineRun from yaml
* [x] PipelineRun
* [ ] TaskRun
* [ ] Run

* Toggle
image
* Editor
image
* Namespace validation
image
* Backend error response handling
image

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (new features, significant UI changes, API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by AlanGreene** [comment]
Thanks for the PR @marniks7. I'm really looking forward to having this functionality added to the Dashboard, and I know it'll make a lot of our users happy.

Overall the approach you've taken here looks good, I have some small suggestions for improvement / a little refactoring but I'll hold off on adding them for now. I have some concerns about the monaco editor, specifically related to its size (can be partly addressed via code splitting), usability on mobile / touch devices, etc. I'm looking into possible solutions / workarounds / alternatives and will post my initial review once I've finished that investigation, should be before Tuesday next week 🤞

**Comment by marniks7** [comment]
@AlanGreene , ok

Some notes:
* For `monaco-editor` it is possible to add autocompletion based on json schema and this will make it fully-featured yaml editor
* For `ace` (https://github.com/ajaxorg/ace), it is not supported https://github.com/ajaxorg/ace/issues/3885

**Review by AlanGreene — CHANGES_REQUESTED** [review]
Thanks again @marniks7.

As I mentioned previously I have some concerns about Monaco and its dependencies:
- it's very heavy, adding ~7MB to our output, essentially quadrupling the size of the static resources to be served to the client
- experience on mobile / touch devices is pretty poor, with frequent overwriting, lines occasionally becoming segmented and difficult to edit in full, backspace / delete across newlines is fairly flaky, and a number of other issues. There doesn't seem to be much progress addressing any of these issues in the community, some of which have been open for 3+ years.

I've reviewed and tested out a number of other editors. So far CodeMirror looks like the only suitable alternative for our current needs. Specifically https://uiwjs.github.io/react-codemirror/. It's significantly lighter (adding ~300Kb to our output compared to Monaco's 7MB!), seems to work quite well on mobile / touch devices, and also supports extensions, completion, etc.

YAML syntax highlighting is provided via https://github.com/codemirror/legacy-modes

Similar to Monaco (and most editors it seems) it uses inline styles so we'll still have to relax our CSP rules to allow `unsafe-inline`. This is unfortunate but should be safe since we have other measures in place that prevent user provided markup / styles from being injected. We should still try to keep it reasonably restrictive, so maybe something like `style-src 'self' 'unsafe-inline'` if that works with it.

CodeMirror isn't quite as full-featured out of the box but seems just as flexible and has a large community and many extensions available.

I've also left a few other comments on the code changes. Let me know what you think.

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ limitations under the License.
 */
 /* istanbul ignore file */
 
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (marniks7) [reply src/containers/CreatePipelineRun/CreatePipelineRun.js:15]: Partially (not that early)

**Comment by marniks7** [comment]
Closed in favor https://github.com/tektoncd/dashboard/pull/2575

**Pull request closed without merging**